docs(admin): openapi custom data provider example (#2116)
api-platform/docs: admin/openapi.md

);
```

### Custom Data Provider

For more advanced use cases, you can create a custom dataProvider to add features such as authentication,
logging, or token-based authorization.

Here's an example of how to integrate a simpleRestProvider with a customized httpClient:

```javascript
import { fetchUtils } from 'react-admin';
import { openApiDataProvider } from '@api-platform/admin';
import simpleRestProvider from 'ra-data-simple-rest';
import { getAccessToken } from './accessToken';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost/api';
const DOC_URL = `${API_URL}/docs`;

// Custom HTTP client for authenticated requests
const httpClient = async (url, options = {}) => {
  options.headers = new Headers({
    ...options.headers,
    Accept: 'application/json',
  });

  const token = getAccessToken();
  if (token) {
    options.user = { token: `Bearer ${token}`, authenticated: true };
  }

  try {
    const { status, headers, body, json } = await fetchUtils.fetchJson(url, options);

>     console.log('HTTP Response:', { status, headers, body, json });

Reviewer: Sure, it would be simpler!

);
```

### Custom Data Provider

For more advanced use cases, you can create a custom dataProvider to add features such as authentication,
logging, or token-based authorization.

> Below is an example of integrating a simpleRestProvider with a custom httpClient:

Reviewer 2: ```suggestion
Here's an example of how to integrate a simpleRestProvider with a customized httpClient:
```

Author: Thanks, will apply the suggested changes.

Pls ask, having someone more experienced here will be beneficial. 
It took me some time to figure it out, so I wanted to update the documentation - to make it easier for other non FE oriented developers.

import { openApiDataProvider } from '@api-platform/admin';
import simpleRestProvider from 'ra-data-simple-rest';
import { getAccessToken } from './accessToken';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost/api';
const DOC_URL = `${API_URL}/docs`;

// Custom HTTP client for authenticated requests
const httpClient = async (url, options = {}) => {
  options.headers = new Headers({
    ...options.headers,
    Accept: 'application/json',
  });

  const token = getAccessToken();
  if (token) {
    options.user = { token: `Bearer ${token}`, authenticated: true };
  }

  try {
    const { status, headers, body, json } = await fetchUtils.fetchJson(url, options);
    console.log('HTTP Response:', { status, headers, body, json });
    return { status, headers, body, json };
  } catch (error) {
    throw error;
  }
};

// Wrapping the custom HTTP client into a data provider
const jsonDataProvider = openApiDataProvider({
  dataProvider: simpleRestProvider(API_URL, httpClient),
  entrypoint: API_URL,
  docEntrypoint: DOC_URL,
});

export default () => (
  <OpenApiAdmin
    entrypoint={API_URL}
    docEntrypoint={DOC_URL}
    dataProvider={jsonDataProvider}
  />
);

```

> > [!NOTE]
> >
> > The `getAccessToken` function is a placeholder for your custom logic to retrieve a JWT token.

Reviewer 2: ```suggestion
> [!NOTE]
> The `getAccessToken` function is a placeholder for your custom logic to retrieve a JWT token.
```

Author: Hey @vinceAmstoutz 

I've applied the suggestions. Please let me know if we can resolve the conversation, and whom I can ask for an additional pair of 👀

Reviewer 2: Sounds good to me @igornast 

I've already asked some colleagues, but now it's time to wait.
